### PR TITLE
Mk nikshay health establishment api

### PR DESCRIPTION
**Pull request by mkangia**
[doc](https://docs.google.com/document/d/1kHEzXLq-8RvPRcRXbYaSNgFKeD3OFP7DR31VytNxxl8/edit#heading=h.h0f9vegd6x4f)

cc: @dannyroberts for Repeater level changes

### CODE REVIEW

**Comment by mkangia** [comment]
@proteusvacuum Trying to cover up the dev work for health establishment integration. I am unsure of quite some payload properties and how the locations are aligned for the private app. Picking up the properties straight from doc AS-IS till i understand those better but otherwise this is how i see this happening in terms of repeater work

File: custom/enikshay/integrations/nikshay/repeater_generator.py [hunk 1]
```diff
@@ -436,6 +441,31 @@ def handle_exception(self, exception, repeat_record):
             )
 
 
+class NikshayHealthEstablishmentPayloadGenerator(LocationPayloadGenerator):
+    def get_payload(self, repeat_record, location):
+        location_hierarchy_codes = get_health_establishment_hierarchy_codes(location)
+        return {
+            'ESTABLISHMENT_TYPE': health_establishment_type.get(location.metadata.get('establishment_type', ''), ''),
```
Comment (dimagimon) [inline custom/enikshay/integrations/nikshay/repeater_generator.py]: E501 line too long (117 > 115 characters)

**Comment by mkangia** [comment]
@proteusvacuum 
i am just left with handling response for the HE Api
i cant do that till i understand what the response would be like
otherwise .. requesting/payload generation …. Location Repeater set up and tests for these is done..

**Review by dannyroberts — APPROVED** [review]
Looks good!

File: custom/enikshay/integrations/nikshay/repeater_generator.py [hunk 2]
```diff
@@ -491,6 +495,21 @@ def get_payload(self, repeat_record, location):
...
+            repeat_record.repeater.operation,
+            response,
+        )
+        message_text = message.find("NewDataSet/HE_DETAILS/Message").text
+        health_facility_id = re.match(HEALTH_ESTABLISHMENT_SUCCESS_RESPONSE_REGEX, message_text).groups()[0]
+        if payload_doc.metadata.get('nikshay_code'):
+            # The repeater checks for this to be absent but in case its added after the trigger
+            # and before its fetched from Nikshay, just keep a copy of the older value for ref
+            payload_doc.metadata['old_nikshay_code'] = payload_doc.metadata.get('nikshay_code')
+        payload_doc.metadata['nikshay_code'] = health_facility_id
```
Comment (proteusvacuum) [inline custom/enikshay/integrations/nikshay/repeater_generator.py]: are you updating the case by doing this?

File: custom/enikshay/integrations/nikshay/repeater_generator.py [hunk 3]
```diff
@@ -491,6 +495,21 @@ def get_payload(self, repeat_record, location):
...
+        payload_doc.save()
```
Comment (proteusvacuum) [inline custom/enikshay/integrations/nikshay/repeater_generator.py]: :+1:

**Review by proteusvacuum — APPROVED** [review]

**Comment by mkangia** [comment]
Ignoring the lint errors due to alignment of xml examples

**Pull request merged**